### ISSUE TITLE
uid inside the List can't be convert to uint32_t

### ISSUE THREAD

**Issue by anisurrahman072** [AgoraIO-Extensions/react-native-agora#755]
**Describe the bug**
A working code broke after almost one year. Fifteen to twenty days ago, the error **‘uid inside the List can’t be converted to uint32_t’** started appearing from the AGORA server when calling the REST API below 👇.

**- REST API:** https://api.agora.io/v1/apps/${AGORA_APP_ID}/cloud_recording/resourceid/${resourceId}/mode/individual/start

**To Reproduce**
Steps to reproduce the behavior. Call the above API from React native app using a fetch() to start Cloud Recording in Individual Mode. Below is the full request body that has been given.

```
await fetch(
		`https://api.agora.io/v1/apps/${AGORA_APP_ID}/cloud_recording/resourceid/${resourceId}/mode/individual/start`,
		{
			headers: {
				Accept: 'application/json',
				'Content-type': 'application/json',
				Authorization: getAgoraAuthorization()
			},
			method: 'post',
			body: JSON.stringify({
				cname: 'db_document_id',
				uid: '69341295',
				clientRequest: {
					recordingConfig: {
						channelType: 1,
						streamTypes: 2,
						maxIdleTime: 30,
						subscribeVideoUids: [hostAgoraUid],
						subscribeAudioUids: [hostAgoraUid],
						subscribeUidGroup: 0
					},
					snapshotConfig: {
						captureInterval: 10,
						fileType: ['jpg']
					},
					storageConfig: {
						vendor: 1,
						region: 0,
						bucket: AWS_S3_BUCKET_NAME,
						accessKey: AWS_ROOT_USER_ACCESS_KEY,
						secretKey: AWS_ROOT_USER_SECRET_KEY,
						fileNamePrefix: [
							AWS_STREAM_RECORDS_FOLDER_NAME,
							userId,
							folderCommonPrefix,
							AWS_STREAM_SNAPSHOTS_FOLDER_NAME
						]
					}
				}
			})
		}
	).then((response) => response.json())
```

**Expected behavior**
I should get a valid response like { sid: "sid_came_from_agora_server", resourceId: "resource_id_came_from_agora_server" }

**Screenshots**
Instead I'm getting like below response (ERROR) 👇
<img width="1403" alt="Screenshot 2023-12-18 at 11 10 50 AM" src="https://github.com/AgoraIO-Extensions/react-native-agora/assets/40535268/f8db5424-0493-4892-886a-4d422beaa052">

**Desktop (please complete the following information):**
 - OS: iOS + Android
 - AGORA SDK Version: "^3.7.0"
 - React native Version: Tested both with RN v0.71.7 & v0.72.0

**Smartphone (please complete the following information):**
 - Device: All devices of IOS & Android

**Additional context**
The code was working smoothly for the last year & **we are a paid user of AGORA**. However, the streaming recording got vulnerable 15-20 days ago and started returning an error message like the one above. We initially thought that the ‘uid’ was causing the issue, but we tried many variations of ‘uid’ in the request body. An example is given below:

As from doc I saw that, uid must be STRING so used these below 👇
- "5239" / "52392" / "523929" / "5239292" / "52392924"

Nothing worked!

 ### As we are a PAID user of AGORA for last 1 year, please keep it in high priority @LichKing-2234

**Comment by guoxianzhe** [comment]
@anisurrahman072 I think this is REST API issue, please open a **technical support ticket** and send the url here. We will contact the relevant technology as soon as possible to follow up on this issue.

**Comment by anisurrahman072** [comment]
Thanks for reaching out. Yea, already created a Ticket with "Integration ISSUE" ✅

**Ticket URL:** https://agora-ticket.agora.io/issue/fb39bb24478d41e2915b3d2047d8d7fd

**Comment by guoxianzhe** [comment]
@anisurrahman072  Thank you, I already contact with tech team and they will contact you from ticket soon.

**Comment by anisurrahman072** [comment]
@guoxianzhe, could you please connect the TICKET with a REACT NATIVE guy, as me & Alex (from AGORA) found that the issue mainly raised from inside AGORA listener. The listener was OK in last month but not sure what happened about it now. 

FYI, the listener code is below

```
_engine.current.addListener('JoinChannelSuccess', async (uid) => 
{
	// Only called when this current USER successfully joined in the channel

	console.log('THE HOST UID came from AGORA: ', uid)

	setHostAgoraUid(String(uid))
})
```

it would be better if you connect the ticket with a REACT NATIVE guy.

**Comment by guoxianzhe** [comment]
@anisurrahman072 Have you upgrade react-native-agora recently?

**Comment by anisurrahman072** [comment]
Not yet upgraded. We are still in SDK version "^3.7.0". As the upgrade process needs some time so made it delay. We used it in more than 30+ screens. Thought to upgrade it when we will feel OK with our feature development.

**Comment by guoxianzhe** [comment]
@anisurrahman072 that is weird, If you didn't upgrade SDK version you should got the same callback result. Maybe The changed part is REST API I think.

**Comment by anisurrahman072** [comment]
Finally solved it. Alex told me to update the listener callback parameters to (channel, uid, ellapsed). That solved it.

So, the code I wrote is almost 1 year old here & also github proves it too. Since then, the code was fully working & our customers streamed & recorded tons of VIDEOS by cloud recording API. Just from this month it went wrong.

After Alex told me, I just updated into 3 parameters & solved it 🔥. Nothing I changed. Only that 3 parameter I added & solved it.

Thanks you guys. We will move to v40.0 for sure soon.
